src/game-flow/game-flow.gateway.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  WebSocketGateway,
+  SubscribeMessage,
+  MessageBody,
+  WebSocketServer,
+  ConnectedSocket,
+} from '@nestjs/websockets';
+import { GameFlowService } from './game-flow.service';
+import { CreateGameFlowDto } from './dto/create-game-flow.dto';
+import { UpdateGameFlowDto } from './dto/update-game-flow.dto';
+import { Server, Socket } from 'socket.io';
+
+@WebSocketGateway({
+  cors: {
+    origin: '*',
+  },
+})
+export class GameFlowGateway {
+  @WebSocketServer()
+  server: Server;
+
+  constructor(private readonly gameFlowService: GameFlowService) {}
+
+  @SubscribeMessage('createGameFlow')
+  async create(
+    @MessageBody() createGameFlowDto: CreateGameFlowDto,
+    @ConnectedSocket() client: Socket,
+  ) {
+    const playersHands = this.gameFlowService.create(
+      createGameFlowDto,
+      client.id,
+    );
+    this.server.emit('initRound', playersHands);
+    return 'playersHands';
+  }
+
+  @SubscribeMessage('findOneGameFlow')
+  findOne(@MessageBody() id: number) {
+    return this.gameFlowService.findOne(id);
+  }
+
+  @SubscribeMessage('flop')
+  flop(@ConnectedSocket() client: Socket) {
+    return this.gameFlowService.flop(client.id);
+  }
+
+  @SubscribeMessage('endRound')
+  endRound() {
+    const commonCards = this.gameFlowService.endRound();
+    this.server.emit('endRound', commonCards);
+    return 'End Round';
+  }
+
+  @SubscribeMessage('river')
+  river() {
+    return this.gameFlowService.river();
+  }
```

Comment:
instead of API for running flop/river, just leave the one for roundEnd. Backend will decide which stage should it run.

In later phase each player could send end round and backend would know what to do next:
- player 1/2 sending end round -> set active player 2
- player 2/2 sending end round -> flop, active player 1
- player 1/2 sending end round -> set active player 2
- player 1/2 sending end round -> turn, set active player 1
... etc.

Later on you can also pass information along end turn message which would contain data about if player bumped up the stake or not. BE will then also decide which turn is it:
- player 1/2 sending end round -> set active player 2
- player 2/2 sending end round with bump -> bumping stake, set active player 1
- player 1/2 sending end round with equal bump -> flop, set active player 1
...